components/trello/sources/new-notification/new-notification.js

```diff
@@ -1,45 +1,44 @@
-const trello = require("../../trello.app.js");
+const common = require("../common-polling.js");
 
 module.exports = {
+  ...common,
   key: "trello-new-notification",
   name: "New Notification",
   description:
     "Emits an event for each new Trello notification for the authenticated user.",
-  version: "0.0.3",
+  version: "0.0.4",
   dedupe: "unique",
-  props: {
-    trello,
-    db: "$.service.db",
-    timer: {
-      type: "$.interface.timer",
-      default: {
-        intervalSeconds: 60 * 15,
-      },
+  methods: {
+    ...common.methods,
+    _getLastNotificationId() {
+      return this.db.get("lastNotificationId") || null;
+    },
+    _setLastNotificationId(notifications) {
+      this.db.set(
+        "lastNotificationId",
+        notifications[notifications.length - 1].id
+      );
+    },
```

Comment:
The setter method should have a single responsibility, which is to persist a notification ID to the DB.
The rest of the logic should be handled in another function (probably the `run()` method, which is the one that usually orchestrates the whole logic).

Reply:
Updated